components/event-publisher-proxy/pkg/handler/http/handler.go

```diff
@@ -136,7 +136,7 @@ func (h *Handler) publishCloudEvents(writer http.ResponseWriter, request *http.R
 	ctx, cancel := context.WithTimeout(request.Context(), h.RequestTimeout)
 	defer cancel()
 
-	message := cehttp.NewMessageFromHttpRequest(request)
+	message := cev2http.NewMessageFromHttpRequest(request)
```

Comment:
why do we:
transform from request to cloudevent
validate cloudevent
transform back to request

when we could just: proxy request, and send response back to sender?

Reply:
the location for this comments is probably not correct. it applies to the entire PR

now that we do not need to use a specific encoding for the cloudevent, we could just proxy the entire thing and rely on that the backend does also some validation before accepting the CE

Reply:
Cause is a major change, it will be solved in #10780

Reply:
i agree, and it is not our main priority now.